src/Server/HTTPHandler.cpp

```diff
@@ -108,6 +109,37 @@ namespace ErrorCodes
     extern const int HTTP_LENGTH_REQUIRED;
 }
 
+namespace
+{
+    /// Process options request. Usefull for CORS.
+    void processOptionsRequest(HTTPServerResponse & response, const Poco::Util::LayeredConfiguration & config)
+    {
+        /// If answer for options request was not defined, return 501 to client.
+        if (!config.has("http_options_response"))
+        {
+            response.setStatusAndReason(HTTPResponse::HTTP_NOT_IMPLEMENTED);
+            response.send();
+        }
+        else
+        {
+            /// otherwise fill response.
+            Strings config_keys;
+            config.keys("http_options_response", config_keys);
+            for (const std::string & config_key : config_keys)
+            {
+                if (config_key == "header" || config_key.starts_with("header["))
+                {
+                    response.add(config.getString("http_options_response." + config_key + ".name", "Empty header"),
```

Comment:
We can avoid default value "Empty header", throwing exception is ok.
Or you can add actually empty header.